src/Microsoft.NetCore.Analyzers/Core/Runtime/UseToLowerInvariantOrToUpperInvariant.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    public abstract class UseToLowerInvariantOrToUpperInvariantAnalyzer : DiagnosticAnalyzer
```

Comment:
How does this rule play with https://github.com/dotnet/roslyn-analyzers/issues/416?

Reply:
This rule only gets activated for `ToUpper` and `ToLower` calls that have no arguments.

The other one(the one you've linked to) gets activated for `ToLower` calls with one argument and `ToLowerInvariant` calls.

The only place where they cross paths is `ToLowerInvariant`. The former recommends using `ToLowerInvariant` instead of `ToLower`.  The latter recommends against using `ToLowerInvariant` but for different reasons.

Basically here's the flow:
`str.ToLower()` => this analyzer => `str.ToLowerInvariant()` => the other analyzer.

Is this a problem?

Reply:
> Basically here's the flow:
`str.ToLower()` => this analyzer => `str.ToLowerInvariant()` => the other analyzer.
Is this a problem?

Yes, we avoid conflict recommendations from analyzers within the repo, i.e. an analyzer should not recommend use `y` instead of `x`, when we know that another analyzer will recommend use `z` instead of `y`, especially if both these rules are turned on by default.

Reply:
Ok, so how should we deal with this particular issue? 
That other analyzer doesn't have a fixer so I'm not even sure what it's meant to do.
Should we drop this issue altogether?

Reply:
Tagging @sharwell @jinujoseph - we should discuss about this rule at the design meeting. Sam/Jinu, how do we want to track this given the issue/PR is in roslyn-analyzers repo.

Reply:
Ping @sharwell @jinujoseph, could you please give us some feedback on this so that the PR can move forward?

Reply:
@sharwell @jinujoseph we need your feedback here.